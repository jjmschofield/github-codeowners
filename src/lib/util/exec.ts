import { exec as realExec } from 'child_process';
import { promisify } from 'util';

export const exec = promisify(realExec);
