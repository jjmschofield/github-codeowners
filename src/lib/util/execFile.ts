import { execFile as realExecFile } from 'child_process';
import { promisify } from 'util';

export const execFile = promisify(realExecFile);
