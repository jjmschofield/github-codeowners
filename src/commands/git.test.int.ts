import { v4 as uuidv4 } from 'uuid';
import fixtures from './__fixtures__/default';
import { generateProject } from './__fixtures__/project-builder.test.helper';

import util from 'util';

const { version } = require('../../package.json');

const exec = util.promisify(require('child_process').exec);

describe('git', () => {
  const testId = uuidv4();

  let testDir = 'not set';

  beforeAll(async () => {
    testDir = await generateProject(testId, fixtures);
    // tslint:disable-next-line:no-console
    console.log(`test scratch dir: ${testDir}`);
  });

  const runCli = async (args: string) => {
    return exec(`node dist/cli.js ${args}`);
  };

  const outputs = ['simple', 'jsonl', 'csv'];

  for (const output of outputs) {
    describe(output, () => {
      it('should list ownership at the specific commit when a commit sha is provided', async () => {
        const { stdout, stderr } = await runCli(`git 2d9bde975c5a5b1a20c57ce0918b0071dcd44e61 -o ${output}`);
        expect(stdout).toMatchSnapshot('stdout');
        expect(stderr).toMatchSnapshot('stderr');
      });

      it('should list ownership of files changed between specific commits', async () => {
        const { stdout, stderr } = await runCli(`git 2d9bde975c5a5b1a20c57ce0918b0071dcd44e61 062f7fe9568b8f66ca97f67c6be9ead0eaba7b38 -o ${output}`);
        expect(stdout).toMatchSnapshot('stdout');
        expect(stderr).toMatchSnapshot('stderr');
      });
    });
  }

  describe('cli', () => {
    it('should print package version', async () => {
      const { stdout, stderr } = await runCli('--version');
      expect(stdout).toEqual(`${version}\n`);
    });
  });
});
