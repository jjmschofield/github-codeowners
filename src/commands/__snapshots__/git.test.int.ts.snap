// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`git csv should list ownership at the specific commit when a commit sha is provided: stderr 1`] = `""`;

exports[`git csv should list ownership at the specific commit when a commit sha is provided: stdout 1`] = `
".github/CODEOWNERS
.gitignore
.idea/codeStyles/codeStyleConfig.xml
.idea/github-codeowners.iml
.idea/inspectionProfiles/Project_Default.xml
.idea/jsLibraryMappings.xml
.idea/misc.xml
.idea/modules.xml
.idea/vcs.xml
README.md
package-lock.json
package.json
src/cli.ts,@jjmschofield
src/commands/audit.ts,@jjmschofield
src/commands/git.ts,@jjmschofield
src/commands/who.ts,@jjmschofield
src/lib/OwnedFile.ts,@jjmschofield
src/lib/OwnershipEngine.ts,@jjmschofield
src/lib/dir.ts,@jjmschofield
src/lib/stats.ts,@jjmschofield
src/lib/types.ts,@jjmschofield
src/lib/writers.ts,@jjmschofield
tsconfig.json
tslint.json
"
`;

exports[`git csv should list ownership of files changed between specific commits: stderr 1`] = `""`;

exports[`git csv should list ownership of files changed between specific commits: stdout 1`] = `
".circleci/config.yml
.gitignore
.idea/runConfigurations/test_int.xml
README.md
__mocks__/recursive-readdir.js
jest.config.js
package-lock.json
package.json
src/cli.ts,@jjmschofield
src/commands/__fixtures__/default/files.json,@jjmschofield
src/commands/__fixtures__/default/gitignore,@jjmschofield
src/commands/__fixtures__/default/gitignore-deep,@jjmschofield
src/commands/__fixtures__/default/index.ts,@jjmschofield
src/commands/__fixtures__/default/owners,@jjmschofield
src/commands/__fixtures__/project-builder.test.helper.ts,@jjmschofield
src/commands/__fixtures__/types.ts,@jjmschofield
src/commands/__snapshots__/audit.test.int.ts.snap,@jjmschofield
src/commands/audit.test.int.ts,@jjmschofield
src/lib/OwnedFile.ts,@jjmschofield
src/lib/OwnershipEngine.test.ts,@jjmschofield
src/lib/OwnershipEngine.ts,@jjmschofield
src/lib/dir.ts,@jjmschofield
src/lib/gitignore.test.ts,@jjmschofield
src/lib/gitignore.ts,@jjmschofield
src/lib/logger.ts,@jjmschofield
src/lib/writers.ts,@jjmschofield
tsconfig.json
tslint.json
"
`;

exports[`git jsonl should list ownership at the specific commit when a commit sha is provided: stderr 1`] = `""`;

exports[`git jsonl should list ownership at the specific commit when a commit sha is provided: stdout 1`] = `
"{\\"path\\":\\".github/CODEOWNERS\\",\\"owners\\":[],\\"lines\\":0}
{\\"path\\":\\".gitignore\\",\\"owners\\":[],\\"lines\\":0}
{\\"path\\":\\".idea/codeStyles/codeStyleConfig.xml\\",\\"owners\\":[],\\"lines\\":0}
{\\"path\\":\\".idea/github-codeowners.iml\\",\\"owners\\":[],\\"lines\\":0}
{\\"path\\":\\".idea/inspectionProfiles/Project_Default.xml\\",\\"owners\\":[],\\"lines\\":0}
{\\"path\\":\\".idea/jsLibraryMappings.xml\\",\\"owners\\":[],\\"lines\\":0}
{\\"path\\":\\".idea/misc.xml\\",\\"owners\\":[],\\"lines\\":0}
{\\"path\\":\\".idea/modules.xml\\",\\"owners\\":[],\\"lines\\":0}
{\\"path\\":\\".idea/vcs.xml\\",\\"owners\\":[],\\"lines\\":0}
{\\"path\\":\\"README.md\\",\\"owners\\":[],\\"lines\\":0}
{\\"path\\":\\"package-lock.json\\",\\"owners\\":[],\\"lines\\":0}
{\\"path\\":\\"package.json\\",\\"owners\\":[],\\"lines\\":0}
{\\"path\\":\\"src/cli.ts\\",\\"owners\\":[\\"@jjmschofield\\"],\\"lines\\":0}
{\\"path\\":\\"src/commands/audit.ts\\",\\"owners\\":[\\"@jjmschofield\\"],\\"lines\\":0}
{\\"path\\":\\"src/commands/git.ts\\",\\"owners\\":[\\"@jjmschofield\\"],\\"lines\\":0}
{\\"path\\":\\"src/commands/who.ts\\",\\"owners\\":[\\"@jjmschofield\\"],\\"lines\\":0}
{\\"path\\":\\"src/lib/OwnedFile.ts\\",\\"owners\\":[\\"@jjmschofield\\"],\\"lines\\":0}
{\\"path\\":\\"src/lib/OwnershipEngine.ts\\",\\"owners\\":[\\"@jjmschofield\\"],\\"lines\\":0}
{\\"path\\":\\"src/lib/dir.ts\\",\\"owners\\":[\\"@jjmschofield\\"],\\"lines\\":0}
{\\"path\\":\\"src/lib/stats.ts\\",\\"owners\\":[\\"@jjmschofield\\"],\\"lines\\":0}
{\\"path\\":\\"src/lib/types.ts\\",\\"owners\\":[\\"@jjmschofield\\"],\\"lines\\":0}
{\\"path\\":\\"src/lib/writers.ts\\",\\"owners\\":[\\"@jjmschofield\\"],\\"lines\\":0}
{\\"path\\":\\"tsconfig.json\\",\\"owners\\":[],\\"lines\\":0}
{\\"path\\":\\"tslint.json\\",\\"owners\\":[],\\"lines\\":0}
"
`;

exports[`git jsonl should list ownership of files changed between specific commits: stderr 1`] = `""`;

exports[`git jsonl should list ownership of files changed between specific commits: stdout 1`] = `
"{\\"path\\":\\".circleci/config.yml\\",\\"owners\\":[],\\"lines\\":0}
{\\"path\\":\\".gitignore\\",\\"owners\\":[],\\"lines\\":0}
{\\"path\\":\\".idea/runConfigurations/test_int.xml\\",\\"owners\\":[],\\"lines\\":0}
{\\"path\\":\\"README.md\\",\\"owners\\":[],\\"lines\\":0}
{\\"path\\":\\"__mocks__/recursive-readdir.js\\",\\"owners\\":[],\\"lines\\":0}
{\\"path\\":\\"jest.config.js\\",\\"owners\\":[],\\"lines\\":0}
{\\"path\\":\\"package-lock.json\\",\\"owners\\":[],\\"lines\\":0}
{\\"path\\":\\"package.json\\",\\"owners\\":[],\\"lines\\":0}
{\\"path\\":\\"src/cli.ts\\",\\"owners\\":[\\"@jjmschofield\\"],\\"lines\\":0}
{\\"path\\":\\"src/commands/__fixtures__/default/files.json\\",\\"owners\\":[\\"@jjmschofield\\"],\\"lines\\":0}
{\\"path\\":\\"src/commands/__fixtures__/default/gitignore\\",\\"owners\\":[\\"@jjmschofield\\"],\\"lines\\":0}
{\\"path\\":\\"src/commands/__fixtures__/default/gitignore-deep\\",\\"owners\\":[\\"@jjmschofield\\"],\\"lines\\":0}
{\\"path\\":\\"src/commands/__fixtures__/default/index.ts\\",\\"owners\\":[\\"@jjmschofield\\"],\\"lines\\":0}
{\\"path\\":\\"src/commands/__fixtures__/default/owners\\",\\"owners\\":[\\"@jjmschofield\\"],\\"lines\\":0}
{\\"path\\":\\"src/commands/__fixtures__/project-builder.test.helper.ts\\",\\"owners\\":[\\"@jjmschofield\\"],\\"lines\\":0}
{\\"path\\":\\"src/commands/__fixtures__/types.ts\\",\\"owners\\":[\\"@jjmschofield\\"],\\"lines\\":0}
{\\"path\\":\\"src/commands/__snapshots__/audit.test.int.ts.snap\\",\\"owners\\":[\\"@jjmschofield\\"],\\"lines\\":0}
{\\"path\\":\\"src/commands/audit.test.int.ts\\",\\"owners\\":[\\"@jjmschofield\\"],\\"lines\\":0}
{\\"path\\":\\"src/lib/OwnedFile.ts\\",\\"owners\\":[\\"@jjmschofield\\"],\\"lines\\":0}
{\\"path\\":\\"src/lib/OwnershipEngine.test.ts\\",\\"owners\\":[\\"@jjmschofield\\"],\\"lines\\":0}
{\\"path\\":\\"src/lib/OwnershipEngine.ts\\",\\"owners\\":[\\"@jjmschofield\\"],\\"lines\\":0}
{\\"path\\":\\"src/lib/dir.ts\\",\\"owners\\":[\\"@jjmschofield\\"],\\"lines\\":0}
{\\"path\\":\\"src/lib/gitignore.test.ts\\",\\"owners\\":[\\"@jjmschofield\\"],\\"lines\\":0}
{\\"path\\":\\"src/lib/gitignore.ts\\",\\"owners\\":[\\"@jjmschofield\\"],\\"lines\\":0}
{\\"path\\":\\"src/lib/logger.ts\\",\\"owners\\":[\\"@jjmschofield\\"],\\"lines\\":0}
{\\"path\\":\\"src/lib/writers.ts\\",\\"owners\\":[\\"@jjmschofield\\"],\\"lines\\":0}
{\\"path\\":\\"tsconfig.json\\",\\"owners\\":[],\\"lines\\":0}
{\\"path\\":\\"tslint.json\\",\\"owners\\":[],\\"lines\\":0}
"
`;

exports[`git simple should list ownership at the specific commit when a commit sha is provided: stderr 1`] = `""`;

exports[`git simple should list ownership at the specific commit when a commit sha is provided: stdout 1`] = `
".github/CODEOWNERS
.gitignore
.idea/codeStyles/codeStyleConfig.xml
.idea/github-codeowners.iml
.idea/inspectionProfiles/Project_Default.xml
.idea/jsLibraryMappings.xml
.idea/misc.xml
.idea/modules.xml
.idea/vcs.xml
README.md
package-lock.json
package.json
src/cli.ts	@jjmschofield
src/commands/audit.ts	@jjmschofield
src/commands/git.ts	@jjmschofield
src/commands/who.ts	@jjmschofield
src/lib/OwnedFile.ts	@jjmschofield
src/lib/OwnershipEngine.ts	@jjmschofield
src/lib/dir.ts	@jjmschofield
src/lib/stats.ts	@jjmschofield
src/lib/types.ts	@jjmschofield
src/lib/writers.ts	@jjmschofield
tsconfig.json
tslint.json
"
`;

exports[`git simple should list ownership of files changed between specific commits: stderr 1`] = `""`;

exports[`git simple should list ownership of files changed between specific commits: stdout 1`] = `
".circleci/config.yml
.gitignore
.idea/runConfigurations/test_int.xml
README.md
__mocks__/recursive-readdir.js
jest.config.js
package-lock.json
package.json
src/cli.ts	@jjmschofield
src/commands/__fixtures__/default/files.json	@jjmschofield
src/commands/__fixtures__/default/gitignore	@jjmschofield
src/commands/__fixtures__/default/gitignore-deep	@jjmschofield
src/commands/__fixtures__/default/index.ts	@jjmschofield
src/commands/__fixtures__/default/owners	@jjmschofield
src/commands/__fixtures__/project-builder.test.helper.ts	@jjmschofield
src/commands/__fixtures__/types.ts	@jjmschofield
src/commands/__snapshots__/audit.test.int.ts.snap	@jjmschofield
src/commands/audit.test.int.ts	@jjmschofield
src/lib/OwnedFile.ts	@jjmschofield
src/lib/OwnershipEngine.test.ts	@jjmschofield
src/lib/OwnershipEngine.ts	@jjmschofield
src/lib/dir.ts	@jjmschofield
src/lib/gitignore.test.ts	@jjmschofield
src/lib/gitignore.ts	@jjmschofield
src/lib/logger.ts	@jjmschofield
src/lib/writers.ts	@jjmschofield
tsconfig.json
tslint.json
"
`;
